src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -118,8 +118,15 @@ public function buildTransitionBlockerList($subject, string $transitionName): Tr
 
             $transitionBlockerList = $this->buildTransitionBlockerListForTransition($subject, $marking, $transition);
 
-            if ($transitionBlockerList->isEmpty()) {
-                continue;
+            // For case of non unique transition names exit the loop on transition that is supposed to be validated.
+            // Else we might get different blocker list containing blockedByMarking instead of blocker from the guard event.
+            // Reason is if you have transition "t" from "a" to "c" and transition "t" from "b" to "c" and you are in place "a"
+            // transitions loop might exit in transition with "from" place "b"
+            // which means you would get $transitionBlockerList containing blockedByMarking instead of guard blocker or none.
+            foreach ($transition->getFroms() as $place) {
+                if ($marking->has($place)) {
+                    break 2;
```

Comment:
What happens if the transition for the current place comes *after* the one one in the list of transitions ?

I think this marking check should happen *before* building the blocker list, and use `continue` rather than `break`

Reply:
There is break instead of continue because if you are making completely invalid transition (transition that exists, but you are not in valid state to execute that transition) then you would not get blocked by marking transition blocker but UndefinedTransitionException.